QuoridorEngine/src/core/QuoridorGame.cs

```diff
@@ -118,6 +155,66 @@ public bool HasWall(int row, int column, Orientation orientation)
             throw new ArgumentException("Unknown orientation type");
         }
 
+        /// <summary>
+        /// Check whether provided wall placement move is valid
+        /// </summary>
+        /// <param name="move">Wall placement move</param>
+        /// <returns>True if valid</returns>
+        public bool CanPlaceWall(QuoridorMove move)
+        {
+            // Check if coordinates are inside bounds
+            if ((move.Column < 0 || move.Column >= dimension - 1) ||
+                (move.Row < 1 || move.Row >= dimension))
+                return false;
+
+            // Check if player has enough walls left
+            if (getTargetPlayer(move.IsWhitePlayer).AvailableWalls <= 0)
+                return false;
+
+            if (move.Orientation == Orientation.Horizontal)
+            {
+                // Check if any walls occupy the space needed by the new
+                // wall or new call forms illegal cross when placed
+                if (board.CheckWallPartHorizontal(move.Row, move.Column) ||
+                    board.CheckWallPartHorizontal(move.Row, move.Column + 1) ||
+                    board.CheckCorner(move.Row, move.Column + 1))
+                    return false;
+            }
+            else if (move.Orientation == Orientation.Vertical)
+            {
+                // Check if any walls occupy the space needed by the new
+                // wall or new call forms illegal cross when placed
+                if (board.CheckWallPartVertical(move.Row, move.Column) ||
+                    board.CheckWallPartVertical(move.Row - 1, move.Column) ||
+                    board.CheckCorner(move.Row, move.Column + 1))
+                    return false;
+            }
+            else
+                // Unknown orientation
+                return false;
+
+            bool whiteTarget = false;
+            bool blackTarget = false;
+
+            Thread white = new(() =>
```

Comment:
I have no clue whether separate threads will work here. Keep in mind that they will both need to continuously access the same data at the same time if they are executed in parallel and this could create conflicts and race conditions